rfcs/ion_1_1/feature-inline_symbols.md

```diff
@@ -0,0 +1,755 @@
+generated identifiers like UUIDs, or content authored by a third party over which a writer has no
+authority. Such values cannot be enumerated in advance and so cannot be added to an SST; in Ion 1.0,
+writing them to the stream requires a Local Symbol Table definition. Using inline symbols, however,
+writers can avoid emitting a symbol table definition no matter what symbols, field names, and
+annotations they encounter in the stream.
+
+Writers have no way of knowing in advance whether the active symbol table contains *all* of the
+symbols, field names, and annotations that they will be asked to write. As such, the full symbol
+discovery phase of writing still has to happen for every value, even if no new symbols will ever be
+found. Inline symbols make it possible to eliminate some of this processing overhead. For example, a
+writer leveraging inline symbols may opt to perform lookups to see if a value is already in the
+symbol table, but avoid the work of adding it to the table if it is not. Alternatively, a writer
+could avoid *all* symbol table lookups for portions of a stream by always writing the text of any
+symbols, field names, and annotations inline.
+
+### Combining inline symbols with Ion templates
+
+When combined with [Ion
+Templates](https://github.com/amzn/ion-docs/blob/ion-templates-rfc/rfcs/ion_templates.md#rfc-ion-templates),
+inline symbols can further reduce the number of symbol IDs that must be added to the symbol table.
+	
+Consider a template representing a class called `com.example.project.Quux`:
+
+```js
+$ion_1_1
+
+// Define a template composed of several symbols:
+
+$ion_symbol_table::{
+  templates: [
+    com.example.project.Quux::{ // Template 1, a struct with 3 blanks
+	  foo: {#0},
+	  bar: {#0},
+	  baz: {#0}
+	}
+  ]
+}
+
+// Invoke template 1 with various different parameters
+
+{#1 1 2 3}
+{#1 4 5 6}
+{#1 7 8 9}
+
+// The above invocations are equivalent to:
+
+com.example.project.Quux::{
+  foo: 1,
+  bar: 2,
+  baz: 3,
+}
+com.example.project.Quux::{
+  foo: 4,
+  bar: 5,
+  baz: 6,
+}
+com.example.project.Quux::{
+  foo: 7,
+  bar: 8,
+  baz: 9,
+}
+```
+
+If we use inline symbol definitions for `foo`, `bar`, `baz`, and `com.example.project.Quux` in our
+template definition, we do not need to add them to the symbol table. By allocating a single template
+ID, we can produce a compact representation of a struct composed of several symbols without growing
+the symbol table.
+
+If the template is frequently used, it can be placed in a Shared Symbol Table.
+
+## Alternatives considered
+
+### Add some new encodings, but not others
+
+For example, add inline symbol structs but not inline symbol values. This would conserve type
+descriptor values and reduce the scope of the changes needed for Ion 1.1. However, it would
+eliminate a key feature for writers.
+
+Inline symbol definitions make it possible for writers to serialize a number of values without
```

Comment:
These use cases should be way up near the start of the document.  (Really the doc should start with use cases and requirements in any case, to make it clear what specific problems are addressed.)